### ISSUE TITLE
Refactor Live Import / External Data frameworks to remove duplicate functionality and simplify code

### ISSUE THREAD

**Issue by kshepherd** [DSpace/DSpace#9758]
The Live Import framework in DSpace is made to return `ImportRecord` objects from input streams (typically transformed from files or HTTP responses), e.g. for populating a new submission.

The External Data framework basically does something similar, but typically the providers are implemented with REST connectors to retrieve remote data (though there is nothing stopping providers from reading files from disk).

Both use spring for bean configuration.

There are several areas where these parts of the code overlap (`MetadatumDTO` vs `MetadataValueDTO`, `ImportRecord` vs `ExternalDataObject`), and in fact there is an ExternalDataProvider impl for LiveImport to 'wrap' it, and there are already comments noting that some of this should be replaced/simplified.

This issue is for discussion around where we can easily make the codebase more simple, some divergences which are harder to solve (or could be blockers for the whole idea), and any gaps in either framework.

My initial proposal would be something like:

* Refactor the really obvious duplicates / overlaps as above so we are always dealing with MetadataValueDTO and ExternalDataObjects in our submission-time live imports
* Use External Data Providers instead of QuerySource? And other similar refactors for live import
* Simplify the external-import/live import system generally, so it's effectively just an interface into External Data framework?
* Provide examples of file-based/stream-based External Data Providers to allow a migration path for existing upload/file based live import implementations
* Fill some extra gaps in External Providers (some more helper methods and spring configs to help with field mapping / transformation, for example
* Extend our IngestionCrosswalk classes to allow an ExternalDataObject to be returned instead of only working with Items/DSOs

Thoughts, comments, questions and criticism welcome!

**Comment by paulo-graca** [comment]
I agree with this need. In the case a local institution needs to extend DSpace with custom external source, it will be in a situation he has to decide or to pick which implementation approach to use. This is a problem in terms of code maintenance. 

But, before we do some code refactoring, I think we can start in the documentation aspect:
https://wiki.lyrasis.org/display/DSDOC8x/Live+Import+from+external+sources
could be changed to be more use-cased oriented with code and configuration examples and how to achieve those.

I think this will help all and prevent the spread of different implementations to achieve the same. Then, we could focus on the code refactoring based on a single way. To me, one of the best implementations here is the Orcid implementation, it's totally configurable, which is good in several ways, but also brings more complexity and configurations spread across DSpace (like: crosswalks, modules, spring).

**Comment by kshepherd** [comment]
Something else I've noticed about LiveImport, is that the map of "MetadataFieldConfig" causes the *field config* to be a unique key, rather than the reference to the MetadataContributor object*. This means you cannot have multiple sets of parsing rules that write the values to the same DC field.

For example consider the datacite-integration spring config:
```
        <entry key-ref="datacite.id" value-ref="dataciteIDContrib"/>
        <entry key-ref="datacite.type" value-ref="dataciteTypesContrib"/>
        <entry key-ref="datacite.creator" value-ref="dataciteAuthorCreator"/>
```

The above keys are enforced unique, and refer to beans defined like so 
```
    <bean id="datacite.id" class="org.dspace.importer.external.metadatamapping.MetadataFieldConfig">
        <constructor-arg value="dc.identifier.doi"/>
    </bean>
```

But the contrib bean itself also takes the 'field' as a parameter
```
    <bean id="dataciteIDContrib" class="org.dspace.importer.external.metadatamapping.contributor.SimpleJsonPathMetadataContributor">
        <property name="field" ref="datacite.id"/>
        <property name="query" value="/doi"/>
    </bean>
```

Which makes more sense... and I feel like we could be simplifying the whole approach to just use a list of Contrib beans which each define a field or fields to which they will write values. Right?

* I also think using the term "Contributor" to mean "Populator" in these classes is hard to follow considering the overloaded usage in DSpace and digital repositories
